infrastructure/prototype-app/template.yaml

```diff
@@ -0,0 +1,16 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+
+Description: "Deploy the Solid proof of concept app"
+
+Resources:
+  ContainerRepository:
+    Type: AWS::ECR::Repository
+    Properties: 
+      ImageTagMutability: IMMUTABLE
```

Comment:
Interesting not come across that one before.
Is that to say you can ship a new image but you can't edit an existing one?

Reply:
Yeah exactly that - see https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-tag-mutability.html

If there's already an image in the repo tagged as `foo-bar`, I can't push another image with that same tag. This PR is using the Git hash as the tag, so I don't see any risk of collision and it prevents someone from getting into our ECR and pushing an image with the deployed tag.